document/js-api/index.bs

```diff
@@ -801,9 +822,37 @@ Assert: |type| is not [=𝗂𝟨𝟦=].
 1. If |type| is [=𝖿𝟨𝟦=],
     1. Let |f64| be ? [=ToNumber=](|v|).
     1. Return [=𝖿𝟨𝟦.𝖼𝗈𝗇𝗌𝗍=] |f64|.
+1. If |type| is [=anyref=],
+    1. If |v| is a primitive value but not a string, symbol, or null, throw |error|.
+    1. Return the result of [=allocating a host address=] for |v|.
+1. If |type| is [=anyeqref=],
+    1. If |v| is a primitive value but not a symbol or null, throw |error|.
+    1. Return the result of [=allocating a host address=] for |v|.
+1. If |type| is [=anyfunc=],
+    1. If |v| is not callable or null, throw |error|.
+    1. Return the result of [=allocating a host address=] for |v|.
+
+</div>
 
+<div algorithm>
+    For <dfn>allocating a host address</dfn> for a value |v|, perform the following steps:
+    1. If |v| is null,
+        1. Return [=ref.null=].
+    1. If |v| is callable,
+        1. If |v| has a \[[FunctionAddress]] internal slot, and therefore is an [=Exported Function=],
+            1. Let |funcaddr| be the value of |v|'s \[[FunctionAddress]] internal slot.
+        1. Otherwise,
+            1. [=Create a host function=] from |v| and let |funcaddr| be the result.
```

Comment:
How is this function's `functype` defined?

Reply:
Ah, excellent catch! I think it simply won't work, since we don't have anything to deduce a type from. So we have to treat regular JS functions as merely opaque anyref objects, not as anyfunc functions that can be called in Wasm. Changed it accordingly (and discovered a [bug](https://github.com/WebAssembly/spec/issues/768) in the JS API draft related to this).